### ISSUE TITLE
Upgrade from 21.0.2 to 21.1.0 fails on oracle db

### ISSUE THREAD

**Issue by Ing0R** [keycloak/keycloak#19841]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the latest release

### Area

storage

### Describe the bug

When updating keycloak 21.0.2 to 21.1.0 the database gets changed. 
jpa-changelog-21.1.0.xml tries to change the dataType of some columns.
This fails with the messages:
```
{​​​​​​​​"timestamp":"2023-04-20T10:56:56.122Z","sequence":8465,"loggerClassName":"org.jboss.logging.Logger","loggerName":"org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler","level":"ERROR","message":"ERROR: Migration failed for changeset META-INF/jpa-changelog-21.1.0.xml::21.1.0-19404::keycloak:\n Reason: liquibase.exception.DatabaseException: ORA-01439: column to be modified must be empty to change datatype\n [Failed SQL: (1439) ALTER TABLE KEYCLOAK_MASTER_TEST.RESOURCE_SERVER_POLICY MODIFY DECISION_STRATEGY NUMBER(3)]","threadName":"main","threadId":1,"mdc":{​​​​​​​​}​​​​​​​​,"ndc":"","hostName":"keycloak-77d7b6c65c-ptd8z","processName":"QuarkusEntryPoint","processId":211}​​​​​​​​
{​​​​​​​​"timestamp":"2023-04-20T10:56:56.122Z","sequence":8466,"loggerClassName":"org.jboss.logging.Logger","loggerName":"org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler","level":"ERROR","message":"ERROR: ORA-01439: column to be modified must be empty to change datatype\n [Failed SQL: (1439) ALTER TABLE KEYCLOAK_MASTER_TEST.RESOURCE_SERVER_POLICY MODIFY DECISION_STRATEGY NUMBER(3)]","threadName":"main","threadId":1,"mdc":{​​​​​​​​}​​​​​​​​,"ndc":"","hostName":"keycloak-77d7b6c65c-ptd8z","processName":"QuarkusEntryPoint","processId":211}​​​​​​​​
{​​​​​​​​"timestamp":"2023-04-20T10:56:56.123Z","sequence":8467,"loggerClassName":"org.jboss.logging.Logger","loggerName":"org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler","level":"ERROR","message":"ERROR: ORA-01439: column to be modified must be empty to change datatype\n","threadName":"main","threadId":1,"mdc":{​​​​​​​​}​​​​​​​​,"ndc":"","hostName":"keycloak-77d7b6c65c-ptd8z","processName":"QuarkusEntryPoint","processId":211}​​​​​​​​
``` 

### Version

21.1.0

### Expected behavior

Update of the column types.

### Actual behavior

Server crash.

### How to Reproduce?

Setup Keycloak 21.0.2 from docker image sourced at quay.io against an oracle database.
Configure it, that the table RESOURCE_SERVER_POLICY gets an entry.
Run Keycloak 21.1.0 against the same database.
View it failing to start.

### Anything else?

In issue https://github.com/keycloak/keycloak/issues/19404 the broken changelog was introduced.

**Comment by ahus1** [comment]
PR available: #19863

**Comment by ahus1** [comment]
Backport to KC21.1 created: #19908
